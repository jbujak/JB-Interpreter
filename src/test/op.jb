int main() {
	print(int_to_string(-1));
	print(int_to_string(1));
	print(int_to_string(1 + 1));
	print(int_to_string(5 - 2));
	print(int_to_string(2 * 2));
	print(int_to_string(10 / 2));
	print(int_to_string(26 % 10));
	print("");

    print(bool_to_string(!false));
    print(bool_to_string(!true));
	print("");

	print(bool_to_string(1 == 1));
	print(bool_to_string(1 == 2));
	print("");

	print(bool_to_string(1 != 2));
	print(bool_to_string(1 != 1));
	print("");

	print(bool_to_string(2 > 1));
	print(bool_to_string(1 > 1));
	print(bool_to_string(0 > 1));
	print("");

	print(bool_to_string(1 < 2));
	print(bool_to_string(1 < 1));
	print(bool_to_string(1 < 0));
	print("");

	print(bool_to_string(2 >= 1));
	print(bool_to_string(1 >= 1));
	print(bool_to_string(0 >= 1));
	print("");

	print(bool_to_string(1 <= 2));
	print(bool_to_string(1 <= 1));
	print(bool_to_string(1 <= 0));
}
