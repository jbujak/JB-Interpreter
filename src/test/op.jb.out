-1
1
2
3
4
5
6

true
false

true
false

true
false

true
false
false

true
false
false

true
true
false

true
true
false
